### PR TITLE
Remove the dependency to SFML.

### PR DESCRIPTION
**Pull request by D8H**

### CODE REVIEW

File: Core/GDCore/String.cpp [hunk 1]
```diff
@@ -112,7 +86,7 @@ String::const_iterator String::end() const
 String String::FromLocale( const std::string &localizedString )
 {
 #if defined(WINDOWS)
-    return FromSfString(sf::String(localizedString)); //Don't need to use the current locale, on Windows, std::locale is always the C locale
+    return FromUTF8(localizedString); //Don't need to use the current locale, on Windows, std::locale is always the C locale
```
Comment (D8H) [inline Core/GDCore/String.cpp:89]: I'm not sure about this.

File: Core/GDCore/String.cpp [hunk 2]
```diff
@@ -164,7 +133,7 @@ String String::FromWide( const std::wstring &wstr )
 std::string String::ToLocale() const
 {
 #if defined(WINDOWS)
-    return ToSfString().toAnsiString();
+    return m_string;
```
Comment (D8H) [inline Core/GDCore/String.cpp:136]: I'm not sure about this either.

File: Core/GDCore/Vector2.h [hunk 3]
```diff
@@ -0,0 +1,360 @@
+// This code comes from the SFML.
```
Comment (D8H) [inline Core/GDCore/Vector2.h:1]: Should it be formatted?

**Comment by D8H** [comment]
On Windows, the UTF8 seams give issues:
```
C:\WorkspaceGDevelop\GDevelop\build-tests\Core>GDCore_tests.exe
terminate called after throwing an instance of 'std::out_of_range'
  what():  [gd::String::substr] starting pos greater than size
```

File: Core/CMakeLists.txt [hunk 4]
```diff
@@ -88,5 +89,5 @@ if(BUILD_TESTS)
 	add_executable(GDCore_tests ${test_source_files})
 	set_target_properties(GDCore_tests PROPERTIES BUILD_WITH_INSTALL_RPATH FALSE) #Allow finding dependencies directly from build path on Mac OS X.
 	target_link_libraries(GDCore_tests GDCore)
-	target_link_libraries(GDCore_tests ${sfml_LIBRARIES})
+	target_link_libraries(GDCore_tests ${CMAKE_DL_LIBS})
```
Comment (D8H) [inline Core/CMakeLists.txt:92]: I had to add this. It was probably done implicitly by the SFML dependency.

File: Core/GDCore/Vector2.h (hunk 3 above)
Comment (4ian) [reply Core/GDCore/Vector2.h:1]: Not mandatory for C++ files (for now).

```suggestion
// This is adapted from SFML (https://github.com/SFML/SFML).
```

File: Core/tests/Utf8Tests.cpp [hunk 5]
```diff
@@ -19,10 +18,8 @@ TEST_CASE("Utf8 String", "[common][utf8]") {
   SECTION("ctor & conversions") {
     gd::String str = u8"UTF8 a été testé !";
 
-    sf::String sfStr = str;
     std::u32string u32str = str.ToUTF32();
 
-    REQUIRE(str == gd::String::FromSfString(sfStr));
     REQUIRE(str == gd::String::FromUTF32(u32str));
   }
```
Comment (4ian) [inline Core/tests/Utf8Tests.cpp:25]: As said in another comment, let's add some tests (check size, substr, assignement from a string to another, append a string to another) with emojis and CJK characters.

**Comment by D8H** [comment]
I tried to add the ansi conversion from SFML, but it doesn't make any difference in the tests and tests.
* ExpressionParser2NaugtyStrings would be interesting to run but it fails at opening the file.
* I thought that the failure on memory consumption was due to the extra string conversion but it seems to act the same without it.

```
C:\WorkspaceGDevelop\GDevelop\build-tests\Core>xcopy /y "C:\WorkspaceGDevelop\GDevelop\Binaries\Output\Release_Windows\GDCore.dll" .
C:\WorkspaceGDevelop\GDevelop\Binaries\Output\Release_Windows\GDCore.dll
1 fichier(s) copié(s)

C:\WorkspaceGDevelop\GDevelop\build-tests\Core>GDCore_tests.exe -e
Start checking naughty strings...

~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
GDCore_tests.exe is a Catch v1.0 b52 host application.
Run with -? for options

-------------------------------------------------------------------------------
ExpressionParser2 - Naughty strings
  Check that no naughty string crash the parser
-------------------------------------------------------------------------------
../Core/tests/ExpressionParser2NaugtyStrings.cpp:16
...............................................................................

../Core/tests/ExpressionParser2NaugtyStrings.cpp:44: FAILED:
explicitly with message:
  Can't open ../Core/tests/ExpressionParser2NaugtyStrings.cpp-blns.txt

-------------------------------------------------------------------------------
EventsList
  Memory consumption
-------------------------------------------------------------------------------
../Core/tests/Common.cpp:29
...............................................................................

../Core/tests/Common.cpp:78: FAILED:
  REQUIRE( 1500 >= endMemory - startMemory )
with expansion:
  1500 >= 2692
with message:
  Memory used: 2692KB

===============================================================================
test cases:   40 |   38 passed | 2 failed
assertions: 3180 | 3178 passed | 2 failed
```

**Comment by 4ian** [comment]
> ExpressionParser2NaugtyStrings would be interesting to run but it fails at opening the file.

Strange, can you try with a hardcoded path? Try with backslash and forward slashes.

> I thought that the failure on memory consumption was due to the extra string conversion but it seems to act the same without it.

I think this test is not relevant on Windows because the memory layouting must be different. ~2-3M of memory consumption is fine. 
This test was rather to protect some kind of memory leak where memory consumption would just to easily 50 or 100M.

> I tried to add the ansi conversion from SFML, but it doesn't make any difference in the tests result.

Thanks for testing! 
If we can somehow run ExpressionParser2NaugtyStrings and the memory test, then everything will run right?
So if the ANSI conversion is not making any change.... let's revert this commit and simplify the code so we just don't do conversions, much like on Linux/macOS :) 

These are tests only, so it only matters that they run. The GDevelop.js version is compiled by Emscripten, so it does not depend if its compiled on Windows, mac or Linux - so we should not introduce any risk of regression.

**Comment by D8H** [comment]
> If we can somehow run ExpressionParser2NaugtyStrings and the memory test, then everything will run right?

I forgot to mention that I disabled tests that check exceptions with `-e` because it seems to make the suite crash on Windows.

**Comment by 4ian** [comment]
Ah nice! What if you try to add a try/catch around them? Otherwise we can remove them or add `#if !defined(WINDOWS)` (or similar) around the REQUIRE_THROWS.

**Comment by D8H** [comment]
About the naughty strings test, the script current folder wasn't right.

I reverted the changes on Ansi conversion. Maybe modern Windows use a standard library that handle UTF8. This script allows to run the tests.
```
xcopy /y "..\Binaries\Output\Release_Windows\GDCore.dll" Core\
"Core\GDCore_tests.exe" -e
PAUSE
```

**Review by 4ian — APPROVED** [review]
🎉
This should be safe if all tests are passing and again we should not have really touched codepaths used by Emscripten for UTF8.
Let's merge this :)

**Pull request merged**